hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/ResourceManager.java

```diff
@@ -1118,38 +1123,36 @@ protected void serviceStop() throws Exception {
     }
   }
 
-    /**
+  /**
    * Transition to standby state in a new thread. The transition operation is
    * asynchronous to avoid deadlock caused by cyclic dependency.
    */
-  private void handleTransitionToStandByInNewThread() {
-    Thread standByTransitionThread =
-        new Thread(activeServices.standByTransitionRunnable);
-    standByTransitionThread.setName("StandByTransitionThread");
-    standByTransitionThread.start();
+  void handleTransitionToStandByInNewThread() {
+    toActiveStandbyExecutor.submit(
+        new RMFatalToStandbyRunner(ResourceManager.getClusterTimeStamp()));
```

Comment:
Based on the spotbugs issue definition, L1131 returns a value that we are not checking. In theory, it may return a bad value and it could go unnoticed.

In this case, you are using ExecutorService.submit() and that could potentially throw NPE or RejectedExecutionException.
The method signature does not specify the `throws` definition. This could be a reason you have the spotbugs issue.

Reference - https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ExecutorService.html#submit(java.util.concurrent.Callable)


Please modify the code here to accommodate the needs of submit().

Reply:
@dineshchitlangia Thank you for your review! Next, I will handle it.